scripts/check_compliance.py

```diff
@@ -304,10 +335,7 @@ def run(self):
 
         if os.path.exists(self.DOCS_WARNING_FILE) and os.path.getsize(self.DOCS_WARNING_FILE) > 0:
             with open(self.DOCS_WARNING_FILE, "rb") as docs_warning:
-                log = docs_warning.read()
-
-                self.case.result = Error("Documentation Issues", "failure")
-                self.case.result._elem.text = log.decode('utf8')
+                self.fail(docs_warning.read().decode("utf-8"))
```

Comment:
Could you use the opportunity to add a new DEFAULT_ENCODING = "utf-8" constant?

Reply:
I prefer the `"utf-8"` here. It'll never be anything else in practice, and probably won't be misspelled without raising an exception either.

Reply:
Windows defaults to UTF-16 in a surprising number of places.
Exception raising assumes 100% test coverage, error handling has notoriously bad coverage - that's where many bugs go and hide.
Duplicated constants are just bad :-)
Anyway that was a bit off-topic, forget it.

Reply:
Won't get UTF-16 for those cases on Windows either.

Not always bad. `decode("utf-8")` gives you more information than `decode(DEFAULT_ENCODING)` a lot quicker, and isn't any more unsafe here in practice.

Personally, I'm big on shaving away as many lines as possible. Bugs hide in clutter. :)